plutus-tx-plugin/test/Blueprint/Tests/Lib.hs

```diff
@@ -135,7 +137,36 @@ newtype Param2b = MkParam2b Bool
 $(PlutusTx.makeLift ''Param2b)
 $(makeIsDataSchemaIndexed ''Param2b [('MkParam2b, 0)])
 
-type Datum2 = Integer
+$( PlutusTx.asData
+    [d|
+      data Datum2 = MkDatum2
+        { datum2integer :: Integer
+        , datum2bool :: Bool
+        }
+        deriving stock (Generic)
+        deriving newtype (AsDefinitionId)
+      |]
+ )
+
+-- The TH splice above generates a 'Datum2' newtype wrapper around the 'BuiltinData'.
+-- Generically derived 'HasSchema Datum2' instance isn't useful as it describes such type
+-- as an opaque 'BuiltinData', so we provide a custom instance to show its structure.
+instance
+  ( HasSchemaDefinition Integer ts
+  , HasSchemaDefinition Bool ts
+  ) =>
+  HasSchema Datum2 ts
```

Comment:
Okay, so you're manually giving an instance here to make the schema machinery work? It does seem like it would be nice if you could just give an instance for the type family that tells you what the child types are, are you sure we can't make that work?

Reply:
Currently the implementation uses a closed type family **with an overlap:**
- Some specific type indexes that don't have a generic instance (`Integer`, `Text`, etc.) -> RHS is defined manually;
- All generic type indexes -> RHS is derived generically.

The moment I try to make it open/associated I hit the "Conflicting family instance declarations" error. 
Related: https://gitlab.haskell.org/ghc/ghc/-/issues/4259

Reply:
Its possible to use an open type family without an overlap if we're willing to give up being able to un-roll any generic type by default. I have demonstrated such alternative in this PR https://github.com/IntersectMBO/plutus/pull/5845

Reply:
Overlapping instances aren't _necessarily_ bad in this case. I think it might work fine if you have `{-# OVERLAPPABLE -#}` on the `Generic` instance, then I think the specific instance should take priority? Or maybe we need to use typeclasses with an associated type for that?

Using a typeclass also has the advantage that the user can then do something like `deriving Unrollable via (Generically MyType)` in the datatype declaration, which looks a little tidier. 

Also, if we're going to expose `Unroll` to the user I think we should give it a more descriptive name. How about `SchemaTypes` or something?